server.js

```diff
@@ -4,12 +4,49 @@ const path = require('path');
 
 const app = express();
 
-// view engine setup
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'hbs');
 app.use(express.static(path.join(__dirname, 'public')));
 
-// add your routes here :)
+app.get('/:code', (req, res) => {
+  const code = req.params.code;
+  const promise1 = fetch('http://localhost:3000/code/' + code)
+  .then(response => response.json());
+  const promise2 = fetch('http://numbersapi.com/' + code)
+  .then(response => response.text());
+
+  Promise.all([promise1, promise2])
+  .then((arr) => {
+    res.render('code', { 
+      'code': arr[0].code, 
+      'phrase': arr[0].phrase,
+      'description': arr[0].description, 
+      'fact': arr[1] 
+    });
+  })
+  .catch(() => {
+    res.send('<h3>Invalid Query, Page not found !<h3>');
+  });
+});
+
+app.get('/', (req, res) => {
+  const rootEndPoint= 'http://localhost:3000/';
+  const badCode = ['203','205','226','407','501','505','308','102','428'];
```

Comment:
True, but if they're never used outside the scope of the function it's best to keep them where it they are... really, whatever makes the program more readable is the route you should take in almost all cases!